### ISSUE TITLE
3.0.0-beta.5: useReferenceInputController permanent filter update issue

### ISSUE THREAD

**Issue by MiMo42** [marmelab/react-admin#3921]
**What you were expecting:**
If you use a ReferenceInput/useReferenceInputController as a dependent component with filter depending on the value of another field, the change of (permanent) filter property should cause an update of data.

**What happened instead:**
useFilterState hook does not detect and handle a change of _permanent_ filter. 

**Steps to reproduce:**
Change filter property of a ReferenceInput e.g. due filtering depending on a value of another field. 

**Related code:**
A suggestion for useFilterState hook to detect changes of permanent filtering:
```
export default ({
  filterToQuery = v => ({ q: v }),
  permanentFilter = {},
  debounceTime = 500
}: UseFilterStateOptions): UseFilterStateProps => {
  const [permFilter, setPermFilter] = useState(permanentFilter);
  const [filter, setFilterValue] = useState({
    ...permanentFilter,
    ...filterToQuery("")
  });

  if (permFilter !== permanentFilter) {
    setPermFilter(permanentFilter);
    setFilterValue({
      ...permanentFilter,
      ...filterToQuery("")
    });
  }

  const setFilter = useCallback(
    debounce(
      value =>
        setFilterValue({
          ...permanentFilter,
          ...filterToQuery(value)
        }),
      debounceTime
    ),
    []
  );

  return {
    filter,
    setFilter
  };
};
```

**Environment**

* React-admin version: 3.0.0-beta.5
* React version: 16.10
* Browser: Chrome

**Comment by Kmaschta** [comment]
Thank you for opening this issue. In order to confirm it, we'll have to reproduce it.
As explained in the bug report template, please fork the following CodeSandbox and repeat your issue on it:

https://codesandbox.io/s/github/marmelab/react-admin/tree/next/examples/simple

This is the simplest way to confirm a bug is related to the React Admin codebase.

**Comment by fzaninotto** [comment]
Fixed by #3928
